File: python/HelloWorld.py
#!/usr/bin/env python

from http.server import BaseHTTPRequestHandler, HTTPServer
from socketserver import ThreadingMixIn
import threading

class Handler(BaseHTTPRequestHandler):

    def do_GET(self):
        self.send_response(200)
        self.send_header('Content-type','text/html')
        self.end_headers()
        self.wfile.write("hello world from python!!".encode())
        return

class ThreadedHTTPServer(ThreadingMixIn, HTTPServer):
    """Handle requests in a separate thread."""

if __name__ == '__main__':
    server = ThreadedHTTPServer(('', 24002), Handler)
    print ('Started server, use <Ctrl-C> to stop')
    server.serve_forever()

File: ruby/src/HelloWorld.rb
#!/usr/bin/env ruby

require 'socket'
server = TCPServer.new 24005

while session = server.accept
  request = session.gets
  puts request

  session.print "HTTP/1.1 200\r\n"
  session.print "Content-Type: text/html\r\n" 
  session.print "\r\n"
  #session.print "Hello world from ruby!!! The time is #{Time.now}"
  session.print "hello world from ruby!!!!"

  session.close
end
